stdlib/json.gr

```diff
@@ -0,0 +1,2103 @@
+              Some(printIndentation) => printIndentation(indentationLevel),
+              None => void,
+            }
+          }
+
+          Buffer.addChar('}', buffer)
+
+          return None
+        },
+        [initialHead, ...initialRest] => {
+          let format = implHelper.format
+
+          Buffer.addChar('{', buffer)
+
+          if (format.objectFormat == OneObjectEntryPerLine) {
+            match (implHelper.printNewLine) {
+              Some(printNewLine) => printNewLine(),
+              None => void,
+            }
+          }
+
+          let mut currentHead = initialHead
+          let mut currentRest = initialRest
+
+          let elemLevel = indentationLevel + 1
+
+          for (let mut index = 0; ; index += 1) {
+            if (index > 0) {
+              Buffer.addChar(',', buffer)
+              if (format.objectFormat == SpacedObjectEntries) {
+                Buffer.addChar(' ', buffer)
+              }
+
+              if (format.objectFormat == OneObjectEntryPerLine) {
+                match (implHelper.printNewLine) {
+                  Some(printNewLine) => printNewLine(),
+                  None => void,
+                }
+              }
+            }
+
+            if (format.objectFormat == OneObjectEntryPerLine) {
+              match (implHelper.printIndentation) {
+                Some(printIndentation) => printIndentation(elemLevel),
+                None => void,
+              }
+            }
+
+            let (key, value) = currentHead
+
+            implHelper.emitEscapedQuotedStringPretty(key)
+
+            Buffer.addChar(':', buffer)
+            match (format.objectFormat) {
+              CompactObjectEntries => void,
+              SpacedObjectEntries | OneObjectEntryPerLine => {
+                Buffer.addChar(' ', buffer)
+              },
+            }
+
+            match (printElementPretty(value, implHelper, elemLevel)) {
+              None => void,
+              err => return err,
+            }
+
+            match (currentRest) {
+              [] => break,
+              [newHead, ...newRest] => {
+                currentHead = newHead
+                currentRest = newRest
+              },
+            }
+          }
+
+          if (format.objectFormat == OneObjectEntryPerLine) {
+            match (implHelper.printNewLine) {
+              Some(printNewLine) => printNewLine(),
+              None => void,
+            }
+            match (implHelper.printIndentation) {
+              Some(printIndentation) => printIndentation(indentationLevel),
+              None => void,
+            }
+          }
+
+          Buffer.addChar('}', buffer)
+
+          return None
+        },
+      }
+    },
+  }
+}
+
+let isCodePointInBasicMultilingualPlane = (code: Number) =>
+  code >= 0x0000 &&
+    code <= 0xFFFF
+
+let isHighSurrogate = (code: Number) => code >= 0xD800 && code <= 0xDBFF
+
+let isLowSurrogate = (code: Number) => code >= 0xDC00 && code <= 0xDFFF
```

Comment:
I thought I saw code above that was doing this too. Do we want to move these higher in the file and use them inside that function?

Reply:
No, I think the code you are reffering to is related to emitting escaped unicode sequences, which does something very different with the surrogates. and the implementations cannot be merged.